kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/KafkaAuthnHandlerTest.java

```diff
@@ -567,6 +571,24 @@ void testCustomRequestFrameAfterSuccessfulAuth() {
         assertEquals(frame, channel.readInbound());
     }
 
+    @Test
+    void testForwardInitialHaProxyMessage() {
+        HAProxyMessage message = new HAProxyMessage(HAProxyProtocolVersion.V1,
+                HAProxyCommand.PROXY, HAProxyProxiedProtocol.TCP4,
+                "1.2.3.4", "5.6.7.8", 65535, 9092);
+        buildChannel(Map.of());
+
+        // Forward initial HaProxyMessage
+        channel.writeInbound(message);
+        assertEquals(message, channel.readInbound());
+
+        // Reject HaProxyMessage if lastSeen not in (START, AUTHN_SUCCESS)
+        kafkaAuthnHandler.lastSeen = KafkaAuthnHandler.State.API_VERSIONS;
+        assertThrows(IllegalStateException.class, () -> channel.writeInbound(message));
+
+        message.release();
+    }
```

Comment:
There are two completely independent assertions here, can we split them into two different test cases.